sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -187,27 +269,88 @@ object GpuCast {
 
   val INVALID_NUMBER_MSG: String = "At least one value is either null or is an invalid number"
 
-  def doCast(
+  private[rapids] def castFloatingTypeToString(input: ColumnView): ColumnVector = {
+    FloatUtils.castFloatingTypeToString(input)
+  }
+
+  def fixDecimalBounds(input: ColumnView,
+      outOfBounds: ColumnView,
+      ansiMode: Boolean): ColumnVector = {
+    if (ansiMode) {
+      withResource(outOfBounds.any()) { isAny =>
+        if (isAny.isValid && isAny.getBoolean) {
+          throw RapidsErrorUtils.arithmeticOverflowError(OVERFLOW_MESSAGE)
+        }
+      }
+      input.copyToColumnVector()
+    } else {
+      withResource(Scalar.fromNull(input.getType)) { nullVal =>
+        outOfBounds.ifElse(nullVal, input)
+      }
+    }
+  }
+
+  def checkNFixDecimalBounds(
+      input: ColumnView,
+      to: DecimalType,
+      ansiMode: Boolean): ColumnVector = {
+    assert(input.getType.isDecimalType)
+    withResource(DecimalUtil.outOfBounds(input, to)) { outOfBounds =>
+      fixDecimalBounds(input, outOfBounds, ansiMode)
+    }
+  }
+
+  def apply(
       input: ColumnView,
       fromDataType: DataType,
       toDataType: DataType,
       ansiMode: Boolean,
       legacyCastToString: Boolean,
       stringToDateAnsiModeEnabled: Boolean): ColumnVector = {
+    new CastOperation(input, fromDataType, toDataType, ansiMode,
```

Comment:
I don't see any value in creating this. An object can implement a trait. We don't need a CastOperation class. It is just adding overhead.